lib/kube/proxy/server.go

```diff
@@ -524,3 +543,55 @@ func (t *TLSServer) setServiceLabels(cluster types.KubeCluster) {
 		cluster.SetDynamicLabels(dstDynLabels)
 	}
 }
+
+// getKubernetesServersForKubeClusterFunc returns a function that returns the kubernetes servers
+// for a given kube cluster depending on the type of service.
+func (t *TLSServer) getKubernetesServersForKubeClusterFunc() (getKubernetesServersForKubeClusterFunc, error) {
+	switch t.KubeServiceType {
+	case KubeService:
+		return func(name string) ([]types.KubeServer, error) {
+			// If this is a kube_service, we can just return the local kube servers.
+			kube, err := t.getKubeClusterForHeartbeat(name)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+			srv, err := types.NewKubernetesServerV3FromCluster(kube, "", t.HostID)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+			return []types.KubeServer{srv}, nil
+		}, nil
+	default:
+		// If not a kube_service, we need to watch the kube_server objects.
+		var err error
+		// Start watching the kube_server objects.
+		t.kubeServerWatcher, err = t.startKubeServerResourceWatcher(t.fwd.ctx)
```

Comment:
To be more specific, why not just use the `AccessPoint` instead of having this watching mechanism?

Reply:
I talked with @espadolini offline about this and it makes sense. Disregard!